client/app/scripts/liveblog-advertising/ads-util.service.js

```diff
@@ -0,0 +1,23 @@
+import angular from 'angular';
+
+export default function adsUtil() {
+	return {
+		uniqueNameInItems: uniqueNameInItems
+	}
+
+	function uniqueNameInItems (item, existingItems) {
+        var invalid = false;
+        if (!item || !item.name) {
+            invalid = gettext('required');
+        }
+        if (!invalid) {
+            angular.forEach(existingItems, function(existingItem) {
```

Comment:
From an answer on stackoverflow, looks like you do this quite a lot more elegantly with es6. See the 4th answer here: http://stackoverflow.com/questions/12462318/find-a-value-in-an-array-of-objects-in-javascript